### ISSUE TITLE
Accessibility review of the cloud.gov Pages public website

### ISSUE THREAD

**Issue by svenaas** [cloud-gov/product#2493]
This task was created as part of the planning task #2471 

Steps:
- [x] Do WAVE scan of each public page in https://cloud.gov/pages/ 
- [x] Assess and remediate any findings of level Error or Contrast Error
- [x] Basic exploration of keyboard navigation of https://cloud.gov/pages/ 
- [x] Basic exploration of screen reader experience of https://cloud.gov/pages/

**Comment by svenaas** [comment]
Starting this in PI 33.6 IP but it will definitely continue into PI 34.1 and probably 34.2 because of my scheduled time out of office.

**Comment by svenaas** [comment]
WAVE scans done; findings noted in cloud-gov/cg-site#2302 and cloud-gov/cg-site#2303

**Comment by svenaas** [comment]
Currently working through keyboard navigation on cloud.gov/pages. So far things seem good and I haven't encountered any navigation traps or issues like that. 

One thing I'm ambivalent about, though, is that in the Documentation section (https://cloud.gov/pages/documentation/ and within) the effect of the skip link — which goes to `#main-content`, assigned to the `<main>` container — is to take users directly into the documentation sidenav, from which they must progress past 37 or so links before they get to the `<h1>` and the actual main content. 

The point of the skip link is to bypass the initial content that's on every page of the site so that users can get straight to the point without repeated navigation or reading of content they've already encountered on previous pages. The Documentation sidenav only appears in the Documentation section, and it's an important navigational element, but once you're in there it's repeated on every page in that section, and it's long. 

I'm not sure why the decision was made to assign `#main-content` on `<main>` instead of on the element containing the `<h1>` (or maybe the question is why the sidenav precedes the `<h1>` instead of being subordinate to it). It might not be difficult to change this, though there could be drawbacks or unanticipated navigational consequences elsewhere in the cloud.gov website. 

Another possibility might be to apply `id='main-content'` to a different element just in the Documentation section, but I suspect this might be impractical, and it would create navigational issues: for one thing, it would make the skip link behavior inconsistent across this site. But more importantly it would mean that users utilizing the skip link once they reach the Documentation section might not encounter the sidenav, which might impose a barrier particularly on blind users. 

We could add a second skip link from the start of the sidenav, linking to the `h1` or its containing element (I'm not sure what good text would be for this — "Skip to section content", maybe?). As a general rule it's not considered a best practice to have multiple skip links — among other reasons, the skip links themselves become a repeated navigational nuisance — but sometimes it's worthwhile ([here's some guidance from WebAIM](https://webaim.org/techniques/skipnav/#multiple)).

I think that last choice — an additional skip link from the sidenav — is the only one of these options worth serious consideration, and the option it's to be weighed against is leaving the currently simpler but repetitive navigation as it is. I'm thinking about it, and welcome others' thoughts.

**Comment by svenaas** [comment]
I've been experimenting. It's easy to add a target ID (in this case to the content div in `docs.html`) but it looks like we're relying on USWDS for the skip link styling (and possibly aspects of the behavior) and the handy USWDS styling of `.usa-skipnav` isn't something we get when applied to a second element further down the DOM. Or at least it's not something I'm getting so far.

**Comment by svenaas** [comment]
OK, the issue was that the "handy styling" consisted of tucking a top left item offscreen at `top: -3.8rem;` until it receives focus and then easing it down into view. This behavior doesn't have the same effect when placed at the top of the sidenav. I played around with adding an override to style a skip link differently at the top of the sidenav and that could definitely work, though I don't think it makes for great UX.

But:

I also discovered that when the viewport is narrower the sidenav flows down below the page content, such that the current skip link behavior takes people all the way past the page content. This is confusing enough, and adding an additional skip link to take people back up is even more confusing than the problem we're starting with.

Ideally I think the skip link should probably take users to the section with the h1, but which section that is varies by layout. Currently the ID is assigned in the `default.html` layout: 

```
  <main id="main-content">
    {{ content }}
  </main>
```

Changing the current state of affairs would involve removing `id="main-content"` from `default.html` and adding that ID to _every_ layout that might get rendered within it. In some layouts the sidenav is conditionally displayed inside that `content` and there is another element to which `#main-content` could be assigned. But in `wide.html` there's just 

```
{{ content }}
```

so it would require adding an extra element.

**Comment by svenaas** [comment]
I've asked in our a11y channel for expert opinions on the skip link. Meanwhile I've moved on to some screen reader  spot-checking, which so far hasn't revealed any issues not already noted (in particular the alt attributes I added following the WAVE scanning addressed what would have been an issue here).

**Comment by svenaas** [comment]
The keyboard navigation issue documented as cloud-gov/cg-site#2320 was resolved with cloud-gov/cg-site#2321.
